### ISSUE TITLE
LT02 - DBT - BigQuery - Formatting Final state inconsistent

### ISSUE THREAD

**Issue by akettmann-apic** [sqlfluff/sqlfluff#7870]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

When formatting an SQL file like this:

```sql
{%- set extraction_list = [
    ('Compound', 'Canonicalized', 'ival'),
    ('IUPAC Name', 'Allowed', 'sval'),
    ('IUPAC Name', 'CAS-like Style', 'sval'),
    ('IUPAC Name', 'Markup', 'sval'),
    ('IUPAC Name', 'Preferred', 'sval'),
    ('IUPAC Name', 'Systematic', 'sval'),
    ('IUPAC Name', 'Traditional', 'sval'),
    ('InChI', 'Standard', 'sval')
] %}

{{ config(cluster_by=['cid']) }}

SELECT
{%- for label, name, field in extraction_list %}
        (
            SELECT
                p.value.{{ field }}
            FROM UNNEST(props) AS p
            WHERE
                p.urn.label = '{{ label }}'
                AND p.urn.name = '{{ name }}'
            LIMIT 1
        )
            AS {{ (label ~ '_' ~ name) | lower | replace(' ', '_') | replace('-','_') }}
        {{- "," if not loop.last }}
{%- endfor %}
FROM {{ ref('pubchem_compound_raw') }}
```

the `{{- "," if not loop.last }}` will be toggled back and forth between 4 indent and 8 indent.



The output that I generated was in an empty directory with no .sqlfluff config file. 

### Expected Behaviour

SQLFluff will format consistently

### Observed Behaviour

run 1
```shell
$ sqlfluff  fix derp.sql --dialect bigquery
==== finding fixable violations ====
== [derp.sql] FAIL                                                                                                                                                                                                                                                                                                                                                                                                                                                           
L:  25 | P:  89 | LT02 | Expected indent of 4 spaces. [layout.indent]
== [derp.sql] FIXED
1 fixable linting violations found
  [2 unfixable linting violations found]
```
run 2
```shell
$ sqlfluff  fix derp.sql --dialect bigquery
==== finding fixable violations ====
== [derp.sql] FAIL                                                                                                                                                                                                                                                                                                                                                                                                                                                           
L:  25 | P:  89 | LT02 | Expected indent of 8 spaces. [layout.indent]
== [derp.sql] FIXED
1 fixable linting violations found
  [2 unfixable linting violations found]
```

### How to reproduce

1. empty directory
2. create derp.sql with contents above provided
3. run sqlfluff with the arguments above provided
4. repeat as many times as desired
5. observe that it swaps between two end states

### Dialect

BigQuery

### Version

$ uv pip freeze | egrep dbt\|sqlfluff
sqlfluff==4.2.1

### Configuration

Empty .sqlfluff and .sqlfluffignore

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by akettmann-apic** [comment]
This occurred first in my env with more requirements, but I found that the problem still occurred when I used a fresh environment with no dbt packages.

**Comment by akettmann-apic** [comment]
sqlfluff fix with -vv

```
$ sqlfluff  fix derp.sql --dialect bigquery -vv
==== sqlfluff ====
sqlfluff:                4.2.1 python:                 3.14.2
implementation:        cpython verbosity:                   2
dialect:              bigquery templater:               jinja
rules:                                all
== Raw Config:
core:                                   
    dialect:            bigquery            
    disable_noqa:       False               
    encoding:           autodetect          
    fix_even_unparsable:False               
    ignore:             []                  
    ignore_templated_areas:True                
    large_file_skip_byte_limit:20000               
    large_file_skip_char_limit:0                   
    large_file_skip_fail:False               
    max_line_length:    80                  
    max_parse_depth:    255                 
    max_parse_nodes:    100000              
    output_line_length: 80                  
    processes:          1                   
    render_variant_limit:5                   
    require_dialect:    False               
    rules:              all                 
    runaway_limit:      10                  
    rust_parser_max_iterations:3000000             
    rust_parser_warn_threshold:2000000             
    sql_file_exts:      .sql,.sql.j2,.dml,.ddl,.pkb
    templater:          jinja               
    use_rust_parser:    auto                
    verbose:            2                   
    warn_unused_ignores:False               
    warnings:           []                  
indentation:                            
    ignore_comment_lines:False               
    implicit_indents:   forbid              
    indent_unit:        space               
    indented_ctes:      False               
    indented_joins:     False               
    indented_on_contents:True                
    indented_then:      True                
    indented_then_contents:True                
    indented_using_on:  True                
    skip_implicit_indents_in:case_expression     
    skip_indentation_in:script_content      
    tab_space_size:     4                   
    template_blocks_indent:True                
    trailing_comments:  before              
layout:                                 
    type:                                   
        array_accessor:                         
            spacing_before:     touch:inline        
        array_type:                             
            spacing_within:     touch:inline        
        assignment_operator:                    
            line_position:      leading             
            spacing_within:     touch               
        binary_operator:                        
            line_position:      leading             
            spacing_within:     touch               
        bind_variable:                          
            spacing_within:     touch               
        bracketed_arguments:                    
            spacing_before:     touch:inline        
        casting_operator:                       
            spacing_after:      touch:inline        
            spacing_before:     touch               
        colon:                                  
            spacing_before:     touch               
        colon_delimiter:                        
            spacing_after:      touch               
            spacing_before:     touch               
        column_path_operator:                    
            line_position:      leading             
            spacing_within:     touch               
        comma:                                  
            line_position:      trailing            
            spacing_before:     touch               
        comment:                                
            spacing_after:      any                 
            spacing_before:     any                 
        common_table_expression:                    
            spacing_within:     single:inline       
        comparison_operator:                    
            line_position:      leading             
            spacing_within:     touch               
        dot:                                    
            spacing_after:      touch               
            spacing_before:     touch               
        end_angle_bracket:                      
            spacing_before:     touch               
        end_bracket:                            
            spacing_before:     touch               
        end_of_file:                            
            spacing_before:     touch               
        end_square_bracket:                     
            spacing_before:     touch               
        from_clause:                            
            line_position:      alone               
        function_contents:                      
            spacing_before:     touch:inline        
        function_name:                          
            spacing_within:     touch:inline        
        function_parameter_list:                    
            spacing_before:     touch:inline        
        groupby_clause:                         
            keyword_line_position:leading             
            line_position:      alone               
        having_clause:                          
            keyword_line_position:leading             
            line_position:      alone               
        join_clause:                            
            line_position:      alone               
        limit_clause:                           
            line_position:      alone               
        match_condition:                        
            spacing_within:     touch:inline        
        numeric_literal:                        
            spacing_within:     touch:inline        
        object_reference:                       
            spacing_within:     touch:inline        
        orderby_clause:                         
            keyword_line_position:leading             
            keyword_line_position_exclusions:
window_specification,
aggregate_order_by,
withingroup_clause,
pipe_operator_clause
            line_position:      leading             
        path_segment:                           
            spacing_within:     touch               
        pattern_expression:                     
            spacing_within:     any                 
        placeholder:                            
            spacing_after:      any                 
            spacing_before:     any                 
        select_clause:                          
            line_position:      alone               
        semi_structured_expression:                    
            spacing_before:     touch:inline        
            spacing_within:     touch:inline        
        set_operator:                           
            line_position:      alone:strict        
        sign_indicator:                         
            spacing_after:      touch:inline        
        sized_array_type:                       
            spacing_within:     touch               
        slash:                                  
            spacing_after:      any                 
            spacing_before:     any                 
        slice:                                  
            spacing_after:      touch               
            spacing_before:     touch               
        sql_conf_option:                        
            spacing_within:     touch               
        sqlcmd_operator:                        
            spacing_before:     touch               
        start_angle_bracket:                    
            spacing_after:      touch               
        start_bracket:                          
            spacing_after:      touch               
        start_square_bracket:                    
            spacing_after:      touch               
        statement_terminator:                    
            line_position:      trailing            
            spacing_before:     touch               
        struct_type:                            
            spacing_within:     touch:inline        
        substitution_variable:                    
            spacing_within:     touch               
        template_loop:                          
            spacing_after:      any                 
            spacing_before:     any                 
        tilde:                                  
            spacing_after:      touch:inline        
        typed_array_literal:                    
            spacing_within:     touch               
        typed_struct_literal:                    
            spacing_within:     touch               
        where_clause:                           
            keyword_line_position:leading             
            keyword_line_position_exclusions:pipe_operator_clause
            line_position:      alone               
rules:                                  
    allow_scalar:       True                
    single_table_references:consistent          
    unquoted_identifiers_policy:all                 
    aliasing.column:                        
        aliasing:           explicit            
    aliasing.forbid:                        
        force_enable:       False               
    aliasing.length:                        
    aliasing.table:                         
        aliasing:           explicit            
    aliasing.unused:                        
        alias_case_check:   dialect             
    ambiguous.column_references:                    
        group_by_and_order_by_style:consistent          
    ambiguous.join:                         
        fully_qualify_join_types:inner               
    capitalisation.functions:                    
        extended_capitalisation_policy:consistent          
    capitalisation.identifiers:                    
        extended_capitalisation_policy:consistent          
    capitalisation.keywords:                    
        capitalisation_policy:consistent          
    capitalisation.literals:                    
        capitalisation_policy:consistent          
    capitalisation.types:                    
        extended_capitalisation_policy:consistent          
    convention.blocked_words:                    
        match_source:       False               
    convention.casting_style:                    
        preferred_type_casting_style:consistent          
    convention.count_rows:                    
        prefer_count_0:     False               
        prefer_count_1:     False               
    convention.not_equal:                    
        preferred_not_equal_style:consistent          
    convention.quoted_literals:                    
        force_enable:       False               
        preferred_quoted_literal_style:consistent          
    convention.select_trailing_comma:                    
        select_clause_trailing_comma:forbid              
    convention.terminator:                    
        multiline_newline:  False               
        require_final_semicolon:False               
    layout.long_lines:                      
        ignore_comment_clauses:False               
        ignore_comment_lines:False               
    layout.newlines:                        
        maximum_empty_lines_between_batches:1                   
        maximum_empty_lines_between_statements:2                   
        maximum_empty_lines_inside_statements:1                   
    layout.select_targets:                    
        single_target_policy:same_line           
        wildcard_policy:    single              
    references.consistent:                    
        force_enable:       False               
    references.from:                        
        force_enable:       False               
    references.keywords:                    
        unquoted_identifiers_policy:aliases             
    references.qualification:                    
        subqueries_ignore_external_references:False               
    references.quoting:                     
        case_sensitive:     True                
        prefer_quoted_identifiers:False               
        prefer_quoted_keyword_style:double_quotes       
        prefer_quoted_keywords:False               
    references.special_chars:                    
        allow_space_in_identifier:False               
        quoted_identifiers_policy:all                 
        unquoted_identifiers_policy:all                 
    structure.join_condition_order:                    
        preferred_first_table_in_join_clause:earlier             
    structure.subquery:                     
        forbid_subquery_in: join                
templater:                              
    unwrap_wrapped_queries:True                
    jinja:                                  
        apply_dbt_builtins: True 
```

**Comment by cblakkan** [comment]
We also run into this for a dbt model, had to downgrade to 4.1.0. Every run just toggles the indentation from 4 to 8 and is always detected as wrong.

```
WITH sns_send_events AS (
  SELECT
        notification_id,
        sns_message_id
    FROM {{ ref('stg_notify__notification_event') }}
    WHERE
        event_type = 'sns_send' AND sns_message_id IS NOT NULL

    {% if is_incremental() %}  -- this line moves from 4 spaces (what it is here) to 8 spaces (to align with the endif)
            AND event_time >= (
                SELECT DATEADD(DAY, -3, MAX(latest_event_at)) FROM {{ this }}  -- noqa: RF02
            )
        {% endif %}

),
```
